api/envoy/admin/v2alpha/certs.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package envoy.admin.v2alpha;
+
+// [#protodoc-title: Certificates]
+
+// Proto representation of certificate details. Admin endpoint uses this wrapper for `/certs` to
+// display certificate information. See :ref:`/certs <operations_admin_interface_certs>` for more
+// information.
+message Certificates {
+  // List of certificates known to an Envoy.
+  repeated Certificate certificates = 1;
+}
+
+message Certificate {
+
+  // Details of CA certificate.
+  CertificateDetails ca_cert = 1;
+
+  // Details of Certificate Chain
+  CertificateDetails cert_chain = 2;
+}
+
+message CertificateDetails {
+  // Path of the certificate.
+  string path = 1;
+
+  // Certificate Serial Number.
+  string serial_number = 2;
+
+  // List of Subject Alternate names.
+  repeated SubjectAlternateName subject_alt_names = 3;
+
+  // Days until expiration of the Certificate.
+  uint64 days_until_expiration = 4;
+}
+
+message SubjectAlternateName {
+  enum Type {
```

Comment:
If I use oneof it would be converted to JSON as
```
    "subject_alt_names": [
     {
      "name": "localhost",
      "DNS": ""
     }
    ],
    "days_until_expiration": "903"
   }
```
I think what we have as below is more readable. Are you suggesting some thing different here?
```
    "subject_alt_names": [
     {
      "name": "localhost",
      "type": "DNS"
     }
    ],
    "days_until_expiration": "903"
   }
```

Reply:
No, I meant you should make this
```
message SubjectAlterativeName {
  oneof name {
    string dns = 1;
    string uri = 2;
  }
}
```

The JSON would be:
```
"subject_alt_names": [
     {
      "dns": "example.com"
     }
]
```

Reply:
Ah.. I see. I will make that change